device_detection.go

```diff
@@ -0,0 +1,243 @@
+	regexp.MustCompile(`(?i)blackberry`),
+	regexp.MustCompile(`(?i)windows phone`),
+}
+
+// DetectDeviceCapabilities analyzes the HTTP request to determine device capabilities
+// Prioritizes User Agent string detection as requested
+func DetectDeviceCapabilities(r *http.Request) DeviceCapabilities {
+	if r == nil {
+		return DeviceCapabilities{
+			SupportsHLS:    false,
+			SupportsMPEGTS: true,
+			IsMobile:       false,
+			IsIOS:          false,
+			IsAndroid:      false,
+			UserAgent:      "",
+			PreferredCodec: "flv",
+		}
+	}
+
+	userAgent := r.Header.Get("User-Agent")
+
+	capabilities := DeviceCapabilities{
+		UserAgent: userAgent,
+	}
+
+	// Primary detection via User Agent string (prioritized as requested)
+	// Detect iOS devices first
+	for _, pattern := range iosPatterns {
+		if pattern.MatchString(userAgent) {
+			capabilities.IsIOS = true
+			break
+		}
+	}
+
+	// Detect Android devices
+	if !capabilities.IsIOS { // Only check if not already detected as iOS
+		for _, pattern := range androidPatterns {
+			if pattern.MatchString(userAgent) {
+				capabilities.IsAndroid = true
+				break
+			}
+		}
+	}
+
+	// Detect mobile devices
+	for _, pattern := range mobilePatterns {
+		if pattern.MatchString(userAgent) {
+			capabilities.IsMobile = true
+			break
+		}
+	}
+
+	// Determine streaming capabilities
+	if capabilities.IsIOS {
+		// iOS devices have native HLS support
+		capabilities.SupportsHLS = true
+		capabilities.SupportsMPEGTS = false // iOS Safari doesn't support MPEG-TS well
+		capabilities.PreferredCodec = "hls"
+	} else if capabilities.IsAndroid {
+		// Android devices may support HLS via hls.js
+		capabilities.SupportsHLS = true
+		capabilities.SupportsMPEGTS = true
+		capabilities.PreferredCodec = "hls" // Prefer HLS for mobile
+	} else {
+		// Desktop browsers - prefer MPEG-TS for better performance
+		capabilities.SupportsHLS = true    // via hls.js
+		capabilities.SupportsMPEGTS = true // via mpegts.js
+		capabilities.PreferredCodec = "flv"
+	}
+
+	return capabilities
+}
+
+// ShouldUseHLS determines if HLS should be used for this request
+func ShouldUseHLS(r *http.Request) bool {
+	if r == nil {
+		return false
+	}
+
+	capabilities := DetectDeviceCapabilities(r)
+
+	// Use HLS for iOS devices as they have native support and better performance
+	if capabilities.IsIOS {
+		return true
+	}
+
+	// Check if explicitly requested via query parameter
+	if r.URL.Query().Get("format") == "hls" {
+		return true
+	}
+
+	// For other devices, use MPEG-TS by default for better performance
+	return false
+}
+
+// GetStreamingFormat returns the preferred streaming format for the device
+func GetStreamingFormat(r *http.Request) string {
+	if ShouldUseHLS(r) {
+		return "hls"
+	}
+	return "flv"
+}
+
+// GetAcceptHeader returns the appropriate Accept header for the streaming format
+func GetAcceptHeader(format string) string {
+	switch format {
+	case "hls":
+		return "application/vnd.apple.mpegurl"
+	case "flv":
```

Comment:
Technically you dont need the "flv" case here when the return equals the default case anyways. You could even argue that the entire switch can be replaced by a conditional `if format=="hls" { return ... } return "video/x-flv"`

Reply:
Original intent was support additional formats. But hls.js made that unnecessary. Will adjust